dot/core/service_test.go

```diff
@@ -117,6 +117,54 @@ func TestAnnounceBlock(t *testing.T) {
 	net.AssertCalled(t, "GossipMessage", expected)
 }
 
+func TestService_InsertKey(t *testing.T) {
+	ks := keystore.NewGlobalKeystore()
+
+	cfg := &Config{
+		Keystore: ks,
+	}
+	s := NewTestService(t, cfg)
+
+	kr, err := keystore.NewSr25519Keyring()
+	require.NoError(t, err)
+
+	testCases := []struct {
+		description  string
+		keystoreType string
+		err          error
+	}{
+		{
+			description:  "Test that insertKey fails when keystore type is invalid ",
+			keystoreType: "some-invalid-type",
+			err:          errors.New("invalid keystore name"),
+		},
+		{
+			description:  "Test that insertKey fails when keystore type is valid but inappropriate",
+			keystoreType: "gran",
+			err:          errors.New("invalid keystore name"),
+		},
+		{
+			description:  "Test that insertKey succeeds when keystore type is valid and appropriate ",
+			keystoreType: "acco",
+			err:          nil,
+		},
+	}
+
+	for _, c := range testCases {
+		c := c
+		t.Run(c.description, func(t *testing.T) {
+			err := s.InsertKey(kr.Alice(), c.keystoreType)
+			require.True(t, errors.Is(err, c.err))
```

Comment:
nit I think there is a `require.IsError` to simplify that

Reply:
There is `require.ErrorIs`, but it won't work. For that, errors have to be in chain.